src/scrabble.js

```diff
@@ -1,5 +1,372 @@
+  const doubleWord = doubleWordChk(str)
+  const tripleWord = tripleWordChk(str)
+  let count1 = 0
+  let count2 = 0
+  let strToCheck = str
+  let newStr = ''
+
+  if (doubleWord.valid === true) {
+    strToCheck = str.slice(1, -1)
+  }
+
+  if (doubleWord.valid === true && tripleWord.valid === true) {
+    strToCheck = str.slice(2, -2)
+  }
+
+  for (let i = 0; i < strToCheck.length; i++) {
+    if (strToCheck[i] === '{') {
+      count1 += 1
+    }
+    if (strToCheck[i + 2] === '}') {
+      count2 += 1
+      // i -= 1
+      newStr += strToCheck[i + 1]
+    }
+  }
+
+  if (count1 === count2) {
+    doubleLettersObj.charsToDouble = newStr
+    doubleLettersObj.valid = true
+  } else {
+    doubleLettersObj.valid = false
+  }
+
+  return doubleLettersObj
+}
+
+const tripleLettersChk = (str) => {
+  const tripleLettersObj = { charsToTriple: '', valid: false }
+  const tripleWord = tripleWordChk(str)
+  const doubleWord = doubleWordChk(str)
+  let count1 = 0
+  let count2 = 0
+  let strToCheck = str
+  let newStr = ''
+
+  if (tripleWord.valid) {
+    strToCheck = str.slice(1, -1)
+  }
+
+  if (doubleWord.valid === true && tripleWord.valid === true) {
+    strToCheck = str.slice(2, -2)
+  }
+
+  for (let i = 0; i < strToCheck.length; i++) {
+    if (strToCheck[i] === '[') {
+      count1 += 1
+    }
+    if (strToCheck[i + 2] === ']') {
+      count2 += 1
+      newStr += strToCheck[i + 1]
+    }
+  }
+  if (count1 === count2) {
+    tripleLettersObj.charsToTriple = newStr
+    tripleLettersObj.valid = true
+  } else {
+    tripleLettersObj.valid = false
+  }
+  return tripleLettersObj
+}
+
+const checkForInvalidChars = (str) => {
+  for (let i = 0; i < str.length; i++) {
+    if (invalidChars.includes(str[i])) {
+      return true
+    }
+  }
+}
+
+const validityChecks = (str) => {
+  const doubleLetters = doubleLettersChk(str)
+  const tripleLetters = tripleLettersChk(str)
+  const invalidChars = checkForInvalidChars(str)
+
+  if (
+    str === '' ||
+    str === null ||
+    str === ' \t\n' ||
+    typeof str !== 'string' ||
+    str === undefined ||
+    invalidChars ||
+    !doubleLetters.valid ||
+    !tripleLetters.valid
+  ) {
+    return false
+  } else {
+    return true
+  }
+}
+
+const calculateDoubleValueLetters = (str) => {
+  const calcDoubledLetters = { doubleLettersPoints: 0 }
+  const lettersToDouble = doubleLettersChk(str)
+  const timesTwo = lettersToDouble.charsToDouble
+  let points = 0
+
+  for (const i of timesTwo) {
+    if (onePoint.includes(i)) {
+      points += 1
+    } else if (twoPoints.includes(i)) {
+      points += 2
+    } else if (threePoints.includes(i)) {
+      points += 3
+    } else if (fourPoints.includes(i)) {
+      points += 4
+    } else if (fivePoints.includes(i)) {
+      points += 5
+    } else if (eightPoints.includes(i)) {
+      points += 8
+    } else if (tenPoints.includes(i)) {
+      points += 10
+    }
+    calcDoubledLetters.doubleLettersPoints = points * 2
```

Comment:
2 is a magic number. Move to a variable that describes it, e.g. `doubleMultiplier`, `tripleMultiplier`